src/dbg/debugger.cpp

```diff
@@ -581,7 +583,42 @@ static void cbGenericBreakpoint(BP_TYPE bptype, void* ExceptionAddress = nullptr
     if(*bp.commandText && commandCondition)  //command
     {
         //TODO: commands like run/step etc will fuck up your shit
-        DbgCmdExec(bp.commandText);
+        varset("$breakpointcondition", breakCondition ? 1 : 0, false);
+        varset("$breakpointlogcondition", logCondition, false);
+        _dbg_dbgcmddirectexec(bp.commandText);

```

Comment:
But if you allow execution of an arbitrary script file anything can happen,
scripts can call commands registered by plugins that change run locks and
so on...
On Sat, 25 Jun 2016 at 18:11, Torusrxxx <notifications@github.com> wrote:

> In src/dbg/debugger.cpp
> <https://github.com/x64dbg/x64dbg/pull/803#discussion_r68490323>:
>
> > @@ -581,7 +583,42 @@ static void cbGenericBreakpoint(BP_TYPE bptype, void* ExceptionAddress = nullptr
> >      if(*bp.commandText && commandCondition)  //command
> >      {
> >          //TODO: commands like run/step etc will fuck up your shit
> > -        DbgCmdExec(bp.commandText);
> > +        varset("$breakpointcondition", breakCondition ? 1 : 0, false);
> > +        varset("$breakpointlogcondition", logCondition, false);
> > +        _dbg_dbgcmddirectexec(bp.commandText);
>
> In fact we shouldn't allow these commands to execute here. We have
> introduced script-controlled condition, so there is no need for such
> commands.
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/x64dbg/x64dbg/pull/803/files/05aadc39a3cf9bc9bc7e55b752342e879af9b42f#r68490323>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ACWCmZCKbPYRgojeKhvTk340qVzEac8Kks5qPVMzgaJpZM4I-X5Q>
> .
>